### DISCUSSION TITLE
Java heap space

### DISCUSSION THREAD

**Discussion by chan18** [dbeaver/dbeaver#12498] [Q&A] [answered]
#### System information: 
![image](https://user-images.githubusercontent.com/13510817/114046469-7d5a6000-98a6-11eb-90e3-c6ed4fc43024.png)
- DBeaver version: Version 21.0.0.202103021012
- Additional extensions: NON

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:
Java heap space

![image](https://user-images.githubusercontent.com/13510817/114046798-c4e0ec00-98a6-11eb-932a-413193929691.png)

#### Steps to reproduce, if exist:
view the database (f4) will show the ER-Digaram of the database. 
save the generated ER-Digram

**Comment by kseniiaguzeeva** [comment]
Please tell your database. Is it embedded? Could you please also send log files https://github.com/dbeaver/dbeaver/wiki/Log-files?
Can be the similar issue with https://github.com/dbeaver/dbeaver/issues/11931

**Reply by FreddyJohn** [reply]
It's massive

**Comment by ITshoushou1** [comment]
![image](https://user-images.githubusercontent.com/22141141/116174879-e1ec3a80-a741-11eb-914a-3142e6a11412.png)
 I got the error too when i export a large table data from database

**Comment by chan18** [comment]
Q) Is it embedded?
A) not sure what you mean by embedded database I assume it is local-DB, I have connected to the remote database.

soon, I will share my logs. Let me try again.
Thanks for your reply @kseniiaguzeeva

**Comment by yhuihu** [comment]
I have the same question.

**Comment by ardelean94** [comment]
I have a similar scenario, I'm trying to connect to a remote Impala DB using DBeaver, I've downloaded and created the jdbc driver but when I try to connect I get an error, see print screen attached
![image](https://user-images.githubusercontent.com/22618053/118320737-532f3a00-b505-11eb-92e7-af24a3948b97.png)

Any ideas?

**Comment by serge-rider** [comment, marked as answer]
@chan18 Perhaps you have a huge schema (hundreds/thousands of tables with many columns).
@ardelean94 Impala JDBC driver needs a lot of memory.

By default DBeaver Community has maximum memory limit 1gb.
You can increase this limit in dbeaver.ini file. 
Try to change `-Xmx1024m` into `-Xmx4096m` to increase maximum memory to 4Gb.

**Reply by zhandshoe** [reply]
I need help finding -Xmx1024m and changing it to -Xmx4096m. I cannot seem to locate the Dbeaver.ini file.

**Reply by rjake** [reply]
for anyone looking, it's under `user\AppData\Local\dbeaver.ini`

**Reply by malhassabi** [reply]
on a mac i found `/Applications/DBeaver.app/Contents/Eclipse/dbeaver.ini`

**Reply by GabrielPontess** [reply]
Thanks, bro!

**Reply by jacobeturpin** [reply]
Know this is an old thread just wanted to give big thanks to @serge-rider. This config change took DBeaver from highly unstable to smooth sailing for me.

**Reply by GSGustavo** [reply]
This worked for me and my friend too. Thanks!

**Reply by Alief321** [reply]
Thank you sir, you are real hero 🔥

**Reply by xtrcode** [reply]
> By default DBeaver Community has maximum memory limit 1gb. You can increase this limit in dbeaver.ini file. Try to change `-Xmx1024m` into `-Xmx4096m` to increase maximum memory to 4Gb.

Sadly that's not possible for the Windows Marketplace version of DBeaver CE, as the installation path is inside the write-protected folder C:\Program Files\WindowsApps

**Comment by chan18** [comment]
hi @serge-rider incressing the ram size from 1gb to 4gb in the dbeaver.ini file fixed my issue.

**Comment by iklobato** [comment]
On Mac
```
vim /Applications/DBeaver.app/Contents/Eclipse/dbeaver.ini
```

**Comment by hungdoit** [comment]
On Linux (Archlinux)
```
/usr/share/dbeaver/dbeaver.ini
```

**Comment by FreddyJohn** [comment]
Yeah, this doesn't help now instead of heap memory failure DBeaver just crashes. Huge database around 4gigs and growing.

**Comment by yoss1209** [comment]
for Linux (Dbeaver community edition)
` /usr/share/dbeaver-ce/dbeaver.ini`
